packages/batch-submitter/src/batch-submitter/tx-batch-submitter.ts

```diff
@@ -432,6 +439,57 @@ export class TransactionBatchSubmitter extends BatchSubmitter {
       return fixedBatch
     }
 
+    const fixSkippedDeposits = async (b: Batch): Promise<Batch> => {
+      this.logger.debug('Fixing skipped deposits...')
+      let nextQueueIndex = await this.chainContract.getNextQueueIndex()
+      const fixedBatch: Batch = []
+      for (const ele of b) {
+        // Look for skipped deposits
+        while (true) {
+          const pendingQueueElements = await this.chainContract.getNumPendingQueueElements()
+          const nextRemoteQueueElements = await this.chainContract.getNextQueueIndex()
+          const totalQueueElements =
+            pendingQueueElements + nextRemoteQueueElements
+          // No more queue elements so we clearly haven't skipped anything
+          if (nextQueueIndex >= totalQueueElements) {
+            break
+          }
+          const [
+            queueEleHash,
+            timestamp,
+            blockNumber,
+          ] = await this.chainContract.getQueueElement(nextQueueIndex)
+
+          if (timestamp < ele.timestamp && blockNumber < ele.blockNumber) {
+            this.logger.error('Fixing skipped deposit', {
+              badTimestamp: ele.timestamp,
+              skippedQueueTimestamp: timestamp,
+              badBlockNumber: ele.blockNumber,
+              skippedQueueBlockNumber: blockNumber,
+            })
+            // Push a dummy queue element
+            fixedBatch.push({
+              stateRoot: ele.stateRoot,
+              isSequencerTx: false,
+              rawTransaction: undefined,
```

Comment:
does it matter that we're setting this to undefined here?

Reply:
`rawTransaction` is only safe to be `undefined` for an enqueue tx which this is. Otherwise the input to the ecdsa contract account will be empty bytes which results in an RLP decoding error (in v0.3.0)